### PR TITLE
Fix GitHub Actions CI

### PR DESCRIPTION
**Pull request by jchodera**
This PR aims to fix the GitHub Actions CI

- [x] Update GitHub Actions workflow
- [x] Test python 3.7 and 3.9 
- [x] Update conda dev environment
- [x] Update code to be compatible with OpenMM 7.6 (without breaking compatibility with earlier versions)
- [x] Fix deprecated `lxml` `Element.getchildren()` method
- [x] Fix temporary directory cleanup

### CODE REVIEW

**Comment by jchodera** [comment]
@yuanqing-wang : I think you might need to sort out the remaining issues here. 

There are some failures due to failure to compare to 8 decimal places that may just need relaxed comparison tolerance, like this:
```
        npt.assert_almost_equal(
>           theta_timemachine, theta_espaloma, decimal=8,
        )
E       AssertionError: 
E       Arrays are not almost equal to 8 decimals
E       
E       Mismatched elements: 999 / 1000 (99.9%)
E       Max absolute difference: 7.01994411e-05
E       Max relative difference: 0.00548936
E        x: array([-2.62901832,  2.6315886 ,  2.96509488, -0.95008304, -0.10972874,
E               1.67166466,  2.96142837,  2.04153377, -0.20819045, -0.11584094,
E              -2.89584473, -0.68454719, -0.7107981 , -0.69531239,  2.01247792,...
E        y: array([-2.62900355,  2.63156933,  2.96509374, -0.95010299, -0.10976114,
E               1.67170121,  2.96142875,  2.04151151, -0.20823212, -0.11582552,
E              -2.8958332 , -0.6845452 , -0.71079845, -0.69533552,  2.01244409,...

espaloma/mm/tests/test_dihedral.py:49: AssertionError
________________________ test_energy_angle_and_bond[g0] ________________________
```

There are some more worrying failures that result in `nan`s, like this:
```
        # test angles
        npt.assert_almost_equal(
            g.nodes["g"].data["u_n3"].numpy(),
            energies["HarmonicAngleForce"],
>           decimal=n_decimals,
        )
E       AssertionError: 
E       Arrays are not almost equal to 3 decimals
E       
E       x and y nan location mismatch:
E        x: array([[37.06]], dtype=float32)
E        y: array(nan)

espaloma/mm/tests/test_openmm_consistency.py:221: AssertionError
----------------------------- Captured stderr call -----------------------------
Warning: : Failed due to unspecified stereochemistry
Warning: Hit maximum duplicate removal time limit for 
Warning: Hit maximum duplicate removal time limit for 
Warning: : Failed due to unspecified stereochemistry
Warning: Hit maximum duplicate removal time limit for 
Warning: Hit maximum duplicate removal time limit for 
------------------------------ Captured log call -------------------------------
```
Given the warnings, I wonder if these are failures of the OpenFF toolkit, which may have introduced new time limits for MCSS matching?

**Comment by jchodera** [comment]
Much more concerningly, these same tests fail locally on my `osx` machine with very different numbers!
```
        # test angles
>       npt.assert_almost_equal(
            g.nodes["g"].data["u_n3"].numpy(),
            energies["HarmonicAngleForce"],
            decimal=n_decimals,
        )
E       AssertionError: 
E       Arrays are not almost equal to 3 decimals
E       
E       Mismatched elements: 1 / 1 (100%)
E       Max absolute difference: 24.394941
E       Max relative difference: 2.16707789
E        x: array([[35.652]], dtype=float32)
E        y: array(11.257)

espaloma/mm/tests/test_openmm_consistency.py:218: AssertionError
```

**Comment by yuanqing-wang** [comment]
@jchodera so what was happening was that the pipeline changed and the XYZ of the simulation was initialized at all zeros. Although the bond geometries and energies can still be computed, the edge case of angle geometry among ((0,0,0), (0,0,0), (0,0,0)) was handled differently with espaloma or with OpenMM with different platforms, hence the different energies or nans. Fixed in https://github.com/choderalab/espaloma/pull/80/commits/173969c0050e103f6532f0ef8b529183bba8a694

**Comment by jchodera** [comment]
@yuanqing-wang : This PR also converts the conda environment into pure conda packages:
https://github.com/choderalab/espaloma/pull/80/files#diff-ba5d22563299549a389183418fe5786b83275382be592bf1ed06fae673b7d086

However, there is not yet a version of `dgl` available from `conda-forge`---it currently has to come from `dglteam`.
This means we may need to try to persuade them to distribute through `conda-forge` at some point.

**Comment by jchodera** [comment]
Looks like there are still a bunch of test failures, but I'm not quite sure what is going wrong here.

**Comment by jchodera** [comment]
@yuanqing-wang : There are still a bunch of failures I don't know how to fix.

File: espaloma/mm/tests/test_openmm_consistency.py [hunk 1]
```diff
@@ -202,7 +202,7 @@ def test_energy_angle_and_bond(g):
 
     # print(g.nodes['n2'].data)
     esp.mm.geometry.geometry_in_graph(g.heterograph)
-    esp.mm.energy.energy_in_graph(g.heterograph, terms=["n2", "n3", "n4", "n4_improper"])
+    esp.mm.energy.energy_in_graph(g.heterograph, terms=["n2", "n3", "n4"])
```
Comment (jchodera) [inline espaloma/mm/tests/test_openmm_consistency.py]: Should we somehow add a note that the `n4_improper` is not implemented correctly, and that we will need to add test coverage later?

**Comment by jchodera** [comment]
@yuanqing-wang : Tests pass now (yay!) so I'll merge this and we can continue additional improvements in new PRs.

**Comment by jchodera** [comment]
Whoops, seems to be an issues with CI failing on startup. I'll open a separate PR to fix.

**Pull request merged**